code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -715,75 +721,119 @@ pub mod pallet {
 			duration_preset: DurationSeconds,
 			keep_alive: bool,
 		) -> Result<Self::PositionId, DispatchError> {
-			let mut rewards_pool =
-				RewardPools::<T>::try_get(pool_id).map_err(|_| Error::<T>::RewardsPoolNotFound)?;
-
-			let reward_multiplier = Self::reward_multiplier(&rewards_pool, duration_preset)
-				.ok_or(Error::<T>::NoDurationPresetsConfigured)?;
+			RewardPools::<T>::try_mutate(pool_id, |maybe_rewards_pool| {
+				let rewards_pool =
+					maybe_rewards_pool.as_mut().ok_or(Error::<T>::RewardsPoolNotFound)?;
+
+				// REVIEW(benluelo): Is this check necessary/ worth it?
+				ensure!(
+					matches!(
+						T::Assets::can_withdraw(*pool_id, who, amount),
+						WithdrawConsequence::Success
+					),
+					Error::<T>::NotEnoughAssets
+				);
 
-			ensure!(
-				matches!(
-					T::Assets::can_withdraw(rewards_pool.asset_id, who, amount),
-					WithdrawConsequence::Success
-				),
-				Error::<T>::NotEnoughAssets
-			);
+				// REVIEW(benluelo): What does "boosted amount" mean?
```

Comment:
This is one of the places where it's sort of obvious what it means. Based on the duration locked the staker gets additional shares in the pool